src/macaron/slsa_analyzer/package_registry/npm_registry.py

```diff
@@ -199,6 +201,47 @@ def download_attestation_payload(self, url: str, download_path: str) -> bool:
 
         return False
 
+    def get_latest_version(self, namespace: str, name: str) -> str | None:
+        """Try to retrieve the latest version of a package from the registry.
+
+        Parameters
+        ----------
+        namespace: str
+            The optional namespace of the package.
```

Comment:
If this `namespace` value is optional, should we make it an optional parameter? Or is there a reason why the type of `namespace` must be `str` instead of `str | None` :thinking: